### PR TITLE
Using upstream Yggradsil

### PR DESCRIPTION
**Pull request by eloycoto**
With this change, flotta-agent will no longer need custom Yggradsil,
and it's using the Upstream version, where a few parameters change name
and it's using TLS if certs are set.

Fix ECOPROJECT-476

To be merged, the following PR should be merged upstream: 
https://github.com/RedHatInsights/yggdrasil/pull/64
https://github.com/RedHatInsights/yggdrasil/pull/65
https://github.com/RedHatInsights/yggdrasil/pull/66

Signed-off-by: Eloy Coto <eloy.coto@acalustra.com>

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -32,7 +32,7 @@ require (
 replace (
 	github.com/containernetworking/cni => github.com/containernetworking/cni v0.8.1
 	github.com/containers/buildah => github.com/containers/buildah v1.23.1
-	github.com/redhatinsights/yggdrasil => github.com/jakub-dzon/yggdrasil v0.0.0-20211012071055-27d969343f4e
+	github.com/redhatinsights/yggdrasil => ../yggdrasil
```
Comment (masayag) [inline go.mod:35]: should yggdrasil version be pinned to a specific one?

Comment (eloycoto) [reply go.mod:35]: yes, yes, as soon my PR are merged we'll pin to the stable version 0.3 :+1:

**Comment by eloycoto** [comment]
@jakub-dzon  your input here will be helpful.

File: internal/datatransfer/gomock_reflect_172602425/prog.go [hunk 2]
```diff
@@ -0,0 +1,66 @@
+
+package main
```
Comment (jakub-dzon) [inline internal/datatransfer/gomock_reflect_172602425/prog.go:2]: I think this file should not be merged.

File: internal/server/deviceserver.go [hunk 3]
```diff
@@ -43,13 +43,16 @@ func (s *deviceServer) Send(ctx context.Context, d *pb.Data) (*pb.Receipt, error
...
 // Disconnect implements the "Disconnect" method of the Worker gRPC service.
-func (s *deviceServer) Disconnect(ctx context.Context, in *pb.Empty) (*pb.DisconnectResponse, error) {
-	log.Infof("received worker disconnect request. DeviceID: %s;", s.configManager.GetDeviceID())
-	err := s.registrationManager.Deregister()
-	if err != nil {
-		log.Warnf("cannot disconnect. DeviceID: %s; err: %v", s.configManager.GetDeviceID(), err)
+func (s *deviceServer) NotifyEvent(ctx context.Context, in *pb.EventNotification) (*pb.EventReceipt, error) {
+	log.Infof("received worker event. DeviceID: %s; Event: %s", s.configManager.GetDeviceID(), in.Name)
+
+	if in.Name == pb.Event_RECEIVED_DISCONNECT || in.Name == pb.Event_UNEXPECTED_DISCONNECT {
```
Comment (jakub-dzon) [inline internal/server/deviceserver.go:49]: Only `RECEIVED_DISCONNECT` should trigger `Deregister` operation; `UNEXPECTED_DISCONNECT` means - "I lost connectivity, but it should come back later".

**Review by jakub-dzon — CHANGES_REQUESTED** [review]
See https://github.com/RedHatInsights/yggdrasil/commit/87ee37bcfd1d5a294496326e8e8c857f2ded326d - the env variable names passed from yggdrasile changed.

**Review by jakub-dzon — APPROVED** [review]

File: flotta-agent.spec [hunk 4]
```diff
@@ -38,12 +38,17 @@ go build -o ./bin/device-worker ./cmd/device-worker
...
 install ./bin/device-worker %{buildroot}%{_libexecdir}/yggdrasil/device-worker
+make install-worker-config LIBEXECDIR=%{_libexecdir} BUILDROOT=%{buildroot} SYSCONFDIR=%{_sysconfdir}
 
 %files
 %{_libexecdir}/yggdrasil/device-worker
+%{_sysconfdir}/yggdrasil/
 
 %changelog
 
+* Fri Feb 11 2022 Eloy Coto <eloycoto@acalustra.com> 1.1
```
Comment (pkliczewski) [inline flotta-agent.spec:49]: Does it mean we want to release this version?

Comment (eloycoto) [reply flotta-agent.spec:49]: No, just to keep track of changes in the RPM specfile.

**Review by machacekondra — APPROVED** [review]

**Review by masayag — APPROVED** [review]

**Pull request merged**